contract/r/gnoswap/gov/governance/proposal.gno

```diff
@@ -430,6 +426,41 @@ func (u *proposalStateUpdater) updateExecutionState() {
 	}
 }
 
+// hasActiveProposal checks if the proposer has any active proposals
+func hasActiveProposal(proposer std.Address) bool {
+	proposalIdRaw, exists := proposalIdListByProposer.Get(proposer.String())
+	if !exists {
+		return false
+	}
+
+	proposalId := proposalIdRaw.(uint64)
+	proposalRaw, exists := proposals.Get(formatUint(proposalId))
+	if !exists {
+		return false
+	}
+
+	proposal := proposalRaw.(ProposalInfo)
+	return proposal.State.Created &&
+		!proposal.State.Canceled &&
+		!proposal.State.Rejected &&
+		!proposal.State.Expired &&
+		!proposal.State.Executed
```

Comment:
need to check `Passed` condition

Reply:
https://github.com/gnoswap-labs/gnoswap/pull/592/commits/8bc11e070eb5dc962da442e8ee38bfb0d1ea1ee1